src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp

```diff
@@ -267,6 +267,54 @@ namespace OpenLoco::World::MapGenerator
         }
     }
 
+    // 0x004611DF
+    static void sub_4611DF()
+    {
+        for (auto pos : World::getDrawableTileRange())
+        {
+            auto tile = TileManager::get(pos);
+            auto* surface = tile.surface();
+
+            if (surface == nullptr)
+                continue;
+
+            if (surface->isIndustrial())
+            {
+                auto* landObj = ObjectManager::get<LandObject>(surface->terrain());
+                if (landObj->hasFlags(LandObjectFlags::unk0))
+                {
+                    if (surface->water())
+                    {
+                        auto waterHeight = surface->waterHeight();
+                        if (surface->slope())
+                            waterHeight -= kSmallZStep;
```

Comment:
water height is in bigZ vanilla used waterZ which is smallZ. So the vanilla -4 has to become -16 i.e. kSmallZStep * kSmallZStep. Alternatively you need to make waterHeight into waterZ and make this -= 4 (not related to kSmallZStep).